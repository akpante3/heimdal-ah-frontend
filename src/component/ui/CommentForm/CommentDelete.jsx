import React from 'react';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';
import { deleteArticleComment } from '../../../actions/CommentActions/CommentActions';
import { deleteCommentReplies } from '../../../actions/ReplyActions/ReplyAction';

/**
  * renderComponent
  * @method Function Function based Component
  * @summary React component for rendering the card components
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing delete comment and replies
  */
const DeleteButton = (props) => {
  const deleteComment = (e) => {
    e.preventDefault();
    const { commentId, buttonName, replyId } = props;
    if (buttonName === 'comment') {
      props.deleteArticleComment(1, commentId);
    } else {
      props.deleteCommentReplies(commentId, replyId);
    }
  };

  /**
     * Renders the component.
     *
     * @memberof app.components.Comment
     * @return {string} - HTML markup for the component
   */

  return (
    <button className='archive_button btn btn-danger' onClick={e => deleteComment(e)} type='submit'><FontAwesome name='trash' className='' /></button>
  );
};

/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  comment: state.comment,
});

export default connect(
  mapStateToProps,
  {
    deleteArticleComment, deleteCommentReplies,
  },
)(DeleteButton);
