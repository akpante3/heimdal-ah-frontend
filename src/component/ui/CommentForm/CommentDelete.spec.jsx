/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { DeleteButton } from './CommentDelete';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      deleteCommentReplies: jest.fn(),
      comment: '',
    };
  });

  it('check the component', () => {
    wrapper = shallow(<DeleteButton {...props} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('should render the component', () => {
    wrapper = shallow(<DeleteButton {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<DeleteButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
