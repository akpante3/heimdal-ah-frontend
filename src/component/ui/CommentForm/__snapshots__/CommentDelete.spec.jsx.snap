// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteButton
    comment=""
    deleteCommentReplies={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FontAwesome
        className=""
        name="trash"
      />,
      "className": "archive_button btn btn-danger",
      "onClick": [Function],
      "type": "submit",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "",
        "name": "trash",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FontAwesome
          className=""
          name="trash"
        />,
        "className": "archive_button btn btn-danger",
        "onClick": [Function],
        "type": "submit",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "name": "trash",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
